src/regex_impl.cc

```diff
@@ -201,12 +201,21 @@ struct RegexParser
         auto it = m_pos.base();
         if (m_regex.end() - it >= 4 and *it++ == '(' and *it++ == '?')
         {
-            auto m = *it++;
-            if ((m != 'i' and m != 'I') or *it++ != ')')
-                return false;
-            m_ignore_case = (m == 'i');
-            m_pos = Iterator{it, m_regex};
-            return true;
+            while (true)
+            {
+                auto m = *it++;
+                if (m == 'i' or m == 'I')
+                    m_ignore_case = (m == 'i');
```

Comment:
Lets move those to a single m_flags (using with_bit_ops) to store all the enabled flags, instead of a boolean for each.